packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -281,6 +286,47 @@ if (__DEV__) {
       });
     }
   };
+
+  ReactStrictModeWarnings.recordLegacyContextWarning = (
+    fiber: Fiber,
+    instance: any,
+  ) => {
+    // Dedup strategy: Warn once per component.
+    if (didWarnAboutLegacyContext.has(fiber.type)) {
+      return;
+    }
+
+    if (
+      typeof instance.getChildContext === 'function' ||
+      fiber.type.contextTypes != null ||
+      fiber.type.childContextTypes != null
+    ) {
+      pendingLegacyContextWarning.push(fiber);
+    }
+  };
+
+  ReactStrictModeWarnings.flushLegacyContextWarning = () => {
+    if (pendingLegacyContextWarning.length > 0) {
+      const uniqueNames = new Set();
+  
+      pendingLegacyContextWarning.forEach((fiber: Fiber) => {
+        didWarnAboutLegacyContext.add(fiber.type);
+        uniqueNames.add(getComponentName(fiber) || 'Component');
+      });
+  
+      const sortedNames = setToSortedString(uniqueNames);
+  
+      lowPriorityWarning(
+        false,
+        'Below are the components that are using legacy context API, ' +
+          'which are subjected to be removed in the future. Please switch to the new ones: ' +
```

Comment:
Couple of thoughts:

* We should probably add the strict root component stack (like we do above with the unsafe lifecycle warning).
* Why low priority warning instead of a warning?
* I'm not sure this warning message matches the tone of our other warnings. Maybe something like...

```js
warning(
  false,
  'Legacy context usage has been detected within a strict-mode tree:%s' +
    '\n\n%s' +
    '\n\nLearn more about this warning here:' +
    '\nhttps://fb.me/react-strict-mode-warnings',
    strictRootComponentStack,
    sortedNames,
);
```

Reply:
What are your thoughts about this proposed wording change, @cyan33 ?

Reply:
Sorry I missed this comment. 

* Consistence wise, I agree. But do we really need to show the component stack? I think the stack information sometimes could easily be noise.
* I'm not quite sure, because I have seen both `lowPriorityWarning` and `warning` in this file and I thought priority-wise this is the same with deprecated lifecycles. But we can definitely change it to `warning`, if it needs more attention from developers.
* I agree that this wording is better. Will change a bit later.